packages/components/src/DataList/DataList.types.ts

```diff
@@ -97,6 +97,21 @@ export interface DataListProps<T extends DataListObject> {
    * The callback function when the user scrolls to the bottom of the list.
    */
   readonly onLoadMore?: () => void;
+
+  /**
+   * Possible actions for selected items in the DataList.
+   */
+  readonly actions?: Record<string, (items: DataListObject["id"][]) => void>;
+
+  /**
+   * Callback when an item row is clicked.
+   */
+  readonly onSelectChange?: (items: DataListObject["id"][]) => void;
```

Comment:
- As per my other comment, rename this to `onSelect`. 
- Should create a type alias for `DataListObject["id"]` since you are using it on multiple places now
- You should also use `T["id"]` since that narrows it down more. 
  - If the consumer has only passed in a `string` as an id...
    - `T["id"]` will be typed as only `string`
    - `DataListObject["id"]` will always be `number | string` regardless

Reply:
I was running into issues relating to consumers with only numbers getting an error of number[] | undefined not compatable with number[] | string[] | undefined. number not compatable with string